yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -545,18 +700,37 @@ func (yb *TargetYugabyteDB) GetListOfTableAttributes(nt sqlname.NameTuple) ([]st
 	return result, nil
 }
 
+const INVALID_INPUT_SYNTAX_ERROR = "invalid input syntax"
+const VIOLATES_UNIQUE_CONSTRAINT_ERROR = "violates unique constraint"
+const SYNTAX_ERROR = "syntax error at"
+const RPC_MSG_LIMIT_ERROR = "Sending too long RPC message"
+
 var NonRetryCopyErrors = []string{
-	"invalid input syntax",
-	"violates unique constraint",
-	"syntax error at",
+	INVALID_INPUT_SYNTAX_ERROR,
+	VIOLATES_UNIQUE_CONSTRAINT_ERROR,
+	SYNTAX_ERROR,
 }
 
 func (yb *TargetYugabyteDB) IsNonRetryableCopyError(err error) bool {
 	NonRetryCopyErrorsYB := NonRetryCopyErrors
-	NonRetryCopyErrorsYB = append(NonRetryCopyErrorsYB, "Sending too long RPC message")
+	NonRetryCopyErrorsYB = append(NonRetryCopyErrorsYB, RPC_MSG_LIMIT_ERROR)
 	return err != nil && utils.ContainsAnySubstringFromSlice(NonRetryCopyErrorsYB, err.Error())
 }
 
+// this function is meant to be used for insert errors during recovery mode
+func (yb *TargetYugabyteDB) IsNonRetryableInsertError(err error, onPrimaryKeyConflictAction string) bool {
```

Comment:
Hey, i don't see this func being used. 
General question: 
- Today, if there is a duplicate PK error, do we retry or immediately fail. Based on the NonRetryCopyErrors, it seems like we retry? (idk if "violates unique constraints" is what we get when there is a duplicate PK)
- Do we plan/want to change that with this PR?

Reply:
Hmm, code from old implementation but this is important. 
Let me think it through once.

Reply:
done. PTAL once.